docs/modules/ROOT/pages/how-tos/worklows/git-submodules.adoc

```diff
@@ -0,0 +1,101 @@
+
+== Implementing this guideline
+=== Basic definitions
+
+- **Upstream repo**: a community project which you do not have push access to (or which you do not want to push CI changes into).
+
+- **Downstream repo**: a self-owned copy of the upstream repository whose build, test, release lifecycle you want to control completely.
+
+=== Create a downstream repo
+
+The downstream repo can be used to do the following:
+
+* hold the git submodule reference to the upstream repository.
+* use the contents of the git submodule in a Containerfile for further edits.
+* onboarded onto {ProductName}.
+
+==== Configuring components
+
+CAUTION: Make sure to properly set the xref:/how-tos/creating.doc#context-and-containerfile-path[Context and Containerfile path] as required for the artifact builds.
+
+See xref:/how-tos/creating.adoc[Creating applications and components] for more details.
+
+==== Configuring the Containerfile
+
+In the Containerfile you can specify the git submodule directories and add any necessary edits that differ from the upstream repository. Here is a basic example:
+
+[source, Dockerfile]
+----
+FROM <base-image>
+
+# Install necessary dependencies
+RUN dnf -y install <dependencies>
+
+# Set the working directory
+ENV <SUBMODULEPKG> /path/to/submodule
+WORKDIR ${<SUBMODULEPKG>}
+
+# Commands to update the submodule for the current image build
+RUN sed -i 's/<old-text>/<new-text>/g' file/in/submodule/directory
+# Commands to build binary from submodule sources
+RUN go build file/in/submodule/directory
+
+WORKDIR /app
+
+LABEL name="<name" \
+      summary="<summary>" \
+      description="<description>"
+
+ENTRYPOINT ["<entrypoint-executable>"]
+----
+
+=== Create the git submodule for the upstream repo to track it
+
+After creating the downstream repo, add a git submodule for the upstream repo:
+
+[source, bash]
+----
+git submodule add <upstream-url>
+----
+
+Check and configure the git submodule config:
+[source, gitmodules]
+----
+# .gitmodules
+[submodule "<project-name>"]
+  path = <path-in-downstream-repo>
+  url = https://github.com/<namespace>/<project>.git
+----
+See the link:https://git-scm.com/docs/gitsubmodules[Git documentation] for more details on configuring git submodules.
+
+**Note**: By default the branch is not specified in the git submodule configuration file when adding a git submodule.
+see link:https://git-scm.com/docs/gitsubmodules[Git documentation] for more details on how to configure the branches for the submodules.
+
+=== Onboard the component onto {ProductName}
+
+After creating the downstream repository you can onboard the component. See xref:/how-tos/creating.adoc[Creating applications and components] for more details.
+
+=== Using private repositories as a git submodule:
+
+When using private repositories we encourage that you give the link:https://github.com/apps/red-hat-konflux[Red Hat {ProductName} Github app] access to that private repository.
```

Comment:
This is only scoped for GitHub. GitLab private submodules are supported as well ... as long as the PAT has access to the repo?

Reply:
What is PAT? I updated the line here and added a sentence for Gitlab repositories: https://github.com/konflux-ci/docs/pull/120/files#diff-8168286ae4fbd44cc1f9e5677aa3a5949ae97f5f391b1466cc253efe6176ccf8R93

Reply:
Personal Access Token: https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html

I assume that GitLab is supported as long as a component would be able to be onboarded (i.e. the proper secret is created). Is this something that you tested?